pkg/storage/ingest/reader.go

```diff
@@ -1076,22 +1070,84 @@ func (r *concurrentFetchers) runFetchers(ctx context.Context, startOffset int64)
 	}
 }
 
-func logCompletedFetch(logger log.Logger, f fetchResult, fetchStartTime time.Time, attempt int, w fetchWant) {
+// handleKafkaFetchErr handles all the errors listed in the franz-go documentation as possible errors when fetching records.
+// For most of them we just apply a backoff. They are listed here so we can be explicit in what we're handling and how.
+// handleKafkaFetchErr returns an adjusted fetchWant in case the error indicated we were consuming not yet produced records or records already deleted due to retention.
+func handleKafkaFetchErr(fr fetchResult, fw fetchWant, shortBackoff, longBackoff *backoff.Backoff, logger log.Logger) fetchWant {
+	err := fr.Err
+	switch {
+	case err == nil:
+	case errors.Is(err, kerr.OffsetOutOfRange):
```

Comment:
I'm not a fan of all this logic here - but I don't have a better suggestion so I'm good with it

Reply:
you mean the one for `OffsetOutOfRange` specifically or the whole switch statement?

Reply:
OffsetOutOfRange specifically.